lib/ReactViews/Story/StoryPanel/StoryBody.tsx

```diff
@@ -45,6 +45,48 @@ const StoryContainer = styled(Box).attrs((props: { isCollapsed: boolean }) => ({
   }
 `;
 
+function shouldAddIframeTag(story: Story) {
+  const parser = new DOMParser();
+  const parsedDocument = parser.parseFromString(story.text, "text/html");
+  const iframes = parsedDocument.getElementsByTagName("iframe");
+  if (iframes.length < 1) return false;
+  let result = true;
+  for (let iframe of iframes) {
+    if (
+      !(
+        iframe.src?.startsWith("https://www.youtube.com/embed/") ||
+        iframe.src?.startsWith("https://www.youtube-nocookie.com/embed/") ||
+        iframe.src?.startsWith("https://player.vimeo.com/video/")
+      )
+    ) {
+      result = false;
+      break;
+    }
+  }
+  return result;
+}
+
+function sourceBasedParse(story: Story) {
+  if (shouldAddIframeTag(story)) {
+    return parseCustomHtmlToReact(
+      story.text,
+      { showExternalLinkWarning: true },
+      false,
+      {
+        ADD_TAGS: ["iframe"],
+        ALLOWED_ATTR: ["src", "width", "height"]
```

Comment:
This will restrict attributes for all tags to only these 3. This will stop e.g. embedded svgs from rendering. See second scene of http://ci.terria.io/issue-6843/#share=s-ejPZyk2EJitArJp0T2naTpnok5Y&playStory=1 compared to the second scene of http://ci.terria.io/main/#share=s-ejPZyk2EJitArJp0T2naTpnok5Y&playStory=1 (the blue svg is not rendered any more).

Reply:
Other examples include text color added using scene editor being ignored

Reply:
Not sure why `svg` tag is removed. Needs investigation.

Reply:
This restriction `ALLOWED_ATTR: ["src", "width", "height"]` does not behave as expected. Removed.